src/server/routes/entity/entity.js

```diff
@@ -519,20 +519,30 @@ module.exports.createEntity = (
 				return model.forge(propsToSet)
 					.save(null, {method: 'insert', transacting});
 			})
-			.then((entityModel) =>
-				achievement.processEdit(req.user.id)
-					.then(() => entityModel)
-			)
 			.then(
 				(entityModel) =>
 					entityModel.refresh({
 						withRelated: ['defaultAlias'],
 						transacting
 					})
 			)
-			.then((entity) => entity.toJSON());
+			.then((entity) =>
+				({
+					entityJSON: entity.toJSON(),
+					editorJSON
+				})
+
+			);
 	});
 
+	entityCreationPromise.then((creationJSON) =>
+		achievement.processEdit(
+			creationJSON.editorJSON.id,
+			creationJSON.entityJSON.revisionId
+		)
+			.then(() => creationJSON.entityJSON)
+	);
+
 	return handler.sendPromiseResult(
 		res,
 		entityCreationPromise,
```

Comment:
The reason I did this was because I need access to the revision in order to get information from it in the achievement system. Moving it elsewhere means the revision is not yet in the database. How would you recommend handling this otherwise?